x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/core/security/transport/netty4/SecurityNetty4TransportTests.java

```diff
@@ -0,0 +1,109 @@
+                .put(TransportSettings.TCP_KEEP_INTERVAL.getKey(), randomIntBetween(-1, 300))
+                .put(TransportSettings.TCP_KEEP_COUNT.getKey(), randomIntBetween(-1, 300))
+                .put(TransportSettings.TCP_SEND_BUFFER_SIZE.getKey(), ByteSizeValue.ofBytes(-1))
+                .put(TransportSettings.TCP_RECEIVE_BUFFER_SIZE.getKey(), ByteSizeValue.ofBytes(-1));
+        }
+        final Settings settings3 = builder3.put(RemoteClusterPortSettings.TCP_NO_DELAY.getKey(), false)
+            .put(RemoteClusterPortSettings.TCP_KEEP_ALIVE.getKey(), false)
+            .put(RemoteClusterPortSettings.TCP_SEND_BUFFER_SIZE.getKey(), ByteSizeValue.ofBytes(42))
+            .put(RemoteClusterPortSettings.TCP_RECEIVE_BUFFER_SIZE.getKey(), ByteSizeValue.ofBytes(99))
+            .put(RemoteClusterPortSettings.TCP_REUSE_ADDRESS.getKey(), false == TransportSettings.TCP_REUSE_ADDRESS.get(Settings.EMPTY))
+            .build();
+        final var options3 = RemoteClusterClientBootStrapOptions.fromSettings(settings3);
+        assertThat(options3.isEmpty(), is(false));
+        assertThat(options3.tcpNoDelay(), is(false));
+        assertThat(options3.tcpKeepAlive(), is(false));
+        assertThat(options3.tcpKeepIdle(), nullValue());
+        assertThat(options3.tcpKeepInterval(), nullValue());
+        assertThat(options3.tcpKeepCount(), nullValue());
+        assertThat(options3.tcpSendBufferSize(), equalTo(ByteSizeValue.ofBytes(42)));
+        assertThat(options3.tcpReceiveBufferSize(), equalTo(ByteSizeValue.ofBytes(99)));
+        assertThat(options3.tcpReuseAddress(), notNullValue());
+
+        // 4. Configure different keepIdle, keepInterval or keepCount
+        final Settings.Builder builder4 = Settings.builder();
+        if (randomBoolean()) {
+            builder4.put(TransportSettings.TCP_NO_DELAY.getKey(), true)
+                .put(TransportSettings.TCP_KEEP_ALIVE.getKey(), true)
+                .put(TransportSettings.TCP_KEEP_IDLE.getKey(), 299)
+                .put(TransportSettings.TCP_KEEP_INTERVAL.getKey(), 299)
+                .put(TransportSettings.TCP_KEEP_COUNT.getKey(), 299)
+                .put(TransportSettings.TCP_SEND_BUFFER_SIZE.getKey(), ByteSizeValue.ofBytes(-1))
+                .put(TransportSettings.TCP_RECEIVE_BUFFER_SIZE.getKey(), ByteSizeValue.ofBytes(-1));
+        }
+        if (randomBoolean()) {
+            builder4.put(RemoteClusterPortSettings.TCP_KEEP_ALIVE.getKey(), true);
+        }
+        final boolean differentKeepIdle = randomBoolean();
+        if (differentKeepIdle) {
+            builder4.put(RemoteClusterPortSettings.TCP_KEEP_IDLE.getKey(), 42);
+        }
+        final boolean differentKeepInterval = randomBoolean();
+        final boolean differentKeepCount = false == differentKeepInterval;
+        if (differentKeepInterval) {
+            builder4.put(RemoteClusterPortSettings.TCP_KEEP_INTERVAL.getKey(), 42);
```

Comment:
Total nit: would make these different values for the different settings, `42`, `43`, `44` to avoid ambiguity